sdk/openai/openai/test/public/utils/utils.ts

```diff
@@ -0,0 +1,202 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { assert } from "vitest";
+import {
+  PipelineRequest,
+  PipelineResponse,
+  createDefaultHttpClient,
+  createEmptyPipeline,
+  createHttpHeaders,
+  createPipelineRequest,
+} from "@azure/core-rest-pipeline";
+import { CognitiveServicesManagementClient } from "@azure/arm-cognitiveservices";
+import { createTestCredential } from "@azure-tools/test-credential";
+import {
+  Recorder,
+  assertEnvironmentVariable,
+  isLiveMode,
+  isRecordMode,
+} from "@azure-tools/test-recorder";
+import {
+  EnvironmentVariableNamesAzureCommon,
+  EnvironmentVariableNamesForAzureSearch,
+  EnvironmentVariableNamesForCompletions,
+  EnvironmentVariableNamesForDalle,
+  EnvironmentVariableNamesForWhisper,
+} from "./envVars.js";
+
+export type AuthMethod = "AAD" | "DummyAPIKey";
+export type DeploymentType = "dalle" | "whisper" | "completions";
+export type APIVersion = "2024-05-01-preview" | "2024-02-01";
```

Comment:
Isn't June GA available?
```suggestion
export type APIVersion = "2024-05-01-preview" | "2024-06-01";
```